### PR TITLE
sendfile can EAGAIN

### PR DESCRIPTION
**Pull request by toolslive**
even when you wait for the socket to become available

### CODE REVIEW

File: ocaml/src/tools/lwt_extra2.ml [hunk 1]
```diff
@@ -255,3 +255,24 @@ let read_file file =
        assert (read = len');
        Lwt.return ()) >>= fun () ->
   Lwt.return buf
+
+
+let copy_using fd_in fd_out size buffer =
```
Comment (domsj) [inline ocaml/src/tools/lwt_extra2.ml]: Hmm, didn't this loop already exist somewhere? (We probably want the code to be shared)

**Comment by domsj** [comment]
looks good

File: ocaml/src/tools/lwt_extra2.ml (hunk 1 above)
Comment (toolslive) [inline ocaml/src/tools/lwt_extra2.ml]: I found a loop like this in the ssl sendfile part, but no where else...

Comment (domsj) [inline ocaml/src/tools/lwt_extra2.ml]: yea, that should be the one

**Pull request merged**